src/zolltools/cache.py

```diff
@@ -0,0 +1,53 @@
+"""Module for caching objects in notebooks"""
+
+from __future__ import annotations
+
+import pickle
+from pathlib import Path
+from collections.abc import Callable
+from typing import TypeVar, Optional, NamedTuple
+
+T = TypeVar("T")
+State = TypeVar("State")
+StorageObject = NamedTuple("StorageObject", [("state", State), ("stored_object", T)])
+
+
+def load(  # pylint: disable=too-many-arguments
+    state: State,
+    generate: Callable[[State], T],
+    unique_id: int,
```

Comment:
I'm currently struggling to figure out the best implementation of `unique_id`. The trick is that if the same `unique_id` and `folder` are entered, they should *always* lead to the same object. This means the method of file path generation should not depend on the session or the surrounding environment.

Maybe something like the following:

```python
folder.joinpath(uuid.UUID(int=hash(str(unique_id))).hex)
```

Reply:
Ended up with the following implementation. Yet to be tested.

```python
if hash_func is None:
    hash_func = hash
assert hash_func is not None

file_path = folder.joinpath(uuid.UUID(int=hash_func(str(unique_id)).hex))
```

Reply:
Note: rejected the use of the [tempfile](https://docs.python.org/3/library/tempfile.html) package because it does not allow for control of the name of the created file. The name of the generated file needs to be the same between function calls.